### PR TITLE
Add a new option to treat an interface like an object

### PR DESCRIPTION
**Pull request by benjaminjkraft**
## Summary:
The basic idea here is if you only request interface fields (no
fragments) you may not care about the concrete type, and so we could
just generate a struct as if it were an object.  I don't think it's a
good idea to do that by default, because then if you later add a
fragment all your code totally changes, but it's quite reasonable as an
option!

Most of the code involved is just wiring and validation; the
core implementation is literally just: treat it like an object.

Issue: https://github.com/Khan/genqlient/issues/85

## Test plan:
make check

### CODE REVIEW

File: docs/FAQ.md [hunk 1]
```diff
@@ -183,6 +183,28 @@ if novel, ok := resp.Favorite.(*GetBooksFavoriteNovel); ok {
...
+
+genqlient will generate just:
+
+```go
+type GetBooksFavoriteBook struct {
+  Title string
+}
+```
+
+Keep in mind that if you later want to add fragments to your selection, the types will change entirely.
```
Comment (csilvers) [inline docs/FAQ.md:205]: Hmm, if you say `struct: true` and specify fragments, how does it work?  Wouldn't it give an error?

**Review by csilvers — COMMENTED** [review]
This is straightforward enough I feel comfortable reviewing it -- it looks good to me!

**Review by StevenACoffman — APPROVED** [review]
Looks great to me!

File: docs/FAQ.md (hunk 1 above)
Comment (benjaminjkraft) [reply docs/FAQ.md:205]: Yeah, I just mean it'll error so you'll have to change it.  Will reword.

**Pull request merged**